frontend/udp/parser.go

```diff
@@ -67,6 +66,7 @@ func ParseAnnounce(r Request, allowIPSpoofing bool) (*bittorrent.AnnounceRequest
 	ip := r.IP
 	ipbytes := r.Packet[84:88]
 	if allowIPSpoofing {
+		// TODO(mrd0ll4r) check if this is an empty IP address?
```

Comment:
Hmm, I feel like the client doesn't know whether we have IP spoofing enabled or not and just sends 0.0.0.0 to tell us "use my public IP", but no idea if that's true or documented anywhere